server-provision.yml

```diff
@@ -115,6 +110,9 @@
           - redis-tools
           - nodejs
           - yarn
+        update_cache: yes
+      register: apt_output
+      changed_when: "not '0 upgraded, 0 newly installed, 0 to remove' in apt_output.stdout"
```

Comment:
I tried running this script second time on a provisioned server, this part fails as the apt_output dictionary doesn't contain this message.

![Screenshot 2020-12-12 at 5 39 54 PM](https://user-images.githubusercontent.com/3954625/101980542-239fab80-3ca1-11eb-8f94-76b69dc22a7e.png)
![Uploading Screenshot 2020-12-12 at 5.30.15 PM.png…]()

Reply:
Hmmm... I thought I had that bit working reliably on both first install and subsequent runs but I thought I got rid of the text search of `stdout`.

I did keep getting myself mixed up in git trying to keep these changes separate from my own changes separate from let's encrypt changes. Based on your result perhaps the `changed_when:` isn't necessary?

Reply:
Ah I see, I tried running the apt part on a provisioned server, Ansible seems smart enough to not run the apt installation step again, and got a "OK" instead of "changed". I think it should be safe to remove the `changed_when` here

Reply:
Cool... do you want me to make those changes?

Also, a referral code for Digital Ocean would be good! I think I've had an account there in the past but it was ages ago and I never used it so no harm trying again.